### ISSUE TITLE
Missing support for keys that are not strings

### ISSUE THREAD

**Issue by anekdoti** [snowflakedb/snowflake-kafka-connector#289]
Currently, the Snowflake Kafka connector stores the key of a message in the field "key" of the METADATA column of the sink table in Snowflake. Furthermore, this only works if the key is deserialized using org.apache.kafka.connect.storage.StringConverter. 
 
This is very limiting, as keys are often structured and e.g. serialized with Avro in the same fashion as the message value.

As the Snowflake Kafka connector already supports Avro deserialization and can use it to represent the message value as the variant column RECORD in the sink table, I don't immediately see a reason why the same could not be done for the message key as well. 

I would propose adding a new variant column KEY to the sink table schema that holds the message key, and adding a second field for the key schema id in the METADATA column. For downwards compatibility reasons, the key field in the METADATA column could persist.

Is there any obstacle for such an extension?

**Comment by raphaelauv** [comment]
Since the bump of the dependency schema-registry (5.5.x ) on the release 1.5.x of the connector it's possible to use a primitive avro string for the key.

```json
  "key.converter.schema.registry.url": "http://my_schema_registry:8081",
  "key.converter": "io.confluent.connect.avro.AvroConverter",
```

But like you said , nothing more complex than this primitive.

**Comment by anekdoti** [comment]
Thank you very much for your answer! 

I think it would be really helpful to support more complex Avro schemas - at least for us, having no support for this really limits the use cases for the Snowflake Kafka connector.

Is there a technical reason why keys are not supported in the same fully general way as values, e.g., in a variant column with all deserializers available in full generality?

For other reasons, I already spent some time with the code of the Snowflake Kafka connector. Would it make sense to try to work on that and create a PR?

**Comment by raphaelauv** [comment]
> For other reasons, I already spent some time with the code of the Snowflake Kafka connector. Would it make sense to try to work on that and create a PR?

I don't know, I do not work for snowflake. Like you I'm just trying to make things work, and that's really not easy !

**Comment by anekdoti** [comment]
I can relate to that :-)  Actually, while trying to find out how to add this feature, I realized that it seems to be implemented in the most recent version 1.5.2 of the connector [by this pull request]{https://github.com/snowflakedb/snowflake-kafka-connector/pull/103). The key is still part of the meta data, but as long as arbitrary Avro is accepted, this is completely fine for me.
